File: 1512-number-of-good-pairs/1512-number-of-good-pairs.scala
object Solution {
  def numIdenticalPairs(nums: Array[Int]): Int = {
    val seen = collection.mutable.Map[Int, Int]()
    var count = 0
    for (n <- nums) {
      if (seen contains n) {
        count = count + seen(n)
        seen(n) = seen(n) + 1
      } else {
        seen(n) = 1
      }
    }
    return count
  }
}

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cs
public class Solution {
  public int NumIdenticalPairs(int[] nums) {
    var dict = new Dictionary<int, int>();
    int ans = 0;
    foreach (int n in nums) {
      if (dict.ContainsKey(n)) {
        ans += dict[n];
        dict[n]++;
      } else {
        dict[n] = 1;
      }
    }
    return ans;
  }
}